### PR TITLE
[12.0][FIX] privacy: Change _default_controller_id() to partner_id.

### PR DESCRIPTION
**Pull request by victoralmau**
Change `_default_controller_id()` to `partner_id`.
This needs to apply to v13 as well.

Please @Yajo and @pedrobaeza can you review it?

@Tecnativa TT31809

### CODE REVIEW

**Comment by pedrobaeza** [comment]
I think it shouldn't be renamed to `company_id`, but keep `controller_id`, and simply change default to `self.env.user.partner_id`.

**Comment by pedrobaeza** [comment]
Now @Yajo should confirm the original intention of the field.

File: privacy/models/privacy_activity.py [hunk 1]
```diff
@@ -49,4 +49,4 @@ class PrivacyActivity(models.Model):
     @api.model
     def _default_controller_id(self):
         """By default it should be the current user's company."""
-        return self.env.user.company_id
+        return self.env.user.partner_id
```
Comment (yajo) [inline privacy/models/privacy_activity.py]: As you see in the docstring, this is incorrect. Instead:

```suggestion
        return self.env.user.company_id.partner_id
```

**Review by yajo — CHANGES_REQUESTED** [review]
The purpose of the controller_id field is to indicate who plays the "controller" role according to GDPR.

It will usually be your company, if you're managing your own privacy stuff. But if you manage your customers' stuff, it would be your customer.

**Review by yajo — APPROVED** [review]

**Pull request merged**